test/test_linalg.py

```diff
@@ -4640,20 +4676,16 @@ def test_matmul_small_brute_force_tunableop(self, device, dtype):
         # disable tunableop buffer rotation for all tests everywhere, it can be slow
         # We set the TunableOp numerical check environment variable here because it is
         # possible to hit some invalid numerical solutions due to the small matrix sizes.
-        # Additionally, we put the entire test in try-finally clause so that
-        # if the test fails/assert, there is no OS environment variabls leaked that
-        # could impact subsequent tests.
         import os
 
-        try:
-            set_tunableop_defaults()
+        tunableop_ctx = self._tunableop_ctx
+
+        with tunableop_ctx():
```

Comment:
Is there a reason this isn't a one-liner?  Would need to fix here and elsewhere.

```suggestion
        with self._tunableop_ctx():
```